### ISSUE TITLE
Handle missing `rev` property on POST /entities/:id

### ISSUE THREAD

**Issue by moll** [Prismatik/auth#22]
Hey,

```js
authDriver.update(purchaser.customerId, {
	permissions: [{type: "purchaser", entity: purchaser.wholesaler}]
})
```

Lead to:
```
Unhandled rejection ReqlRuntimeError: Cannot convert `undefined` with r.expr() in:
r.table("entities").get("9b061055-e06a-460f-a068-370d54416af1")("rev").eq(undefined)
                                                                          ^^^^^^^^^

    at Term.run (/home/user/Documents/auth/node_modules/rethinkdbdash/lib/term.js:43:17)
    at Server.exports.validateEmail.exports.validateRevision.r.table.get.getField.eq.run.then (/home/user/Documents/auth/lib/entity.js:31:4)
    at next (/home/user/Documents/auth/node_modules/restify/lib/server.js:906:30)
    at f (/home/user/Documents/auth/node_modules/restify/node_modules/once/once.js:17:25)
    at Server.exports.validateEmail.query.isEmpty.run.then (/home/user/Documents/auth/lib/entity.js:9:23)
    at next (/home/user/Documents/auth/node_modules/restify/lib/server.js:906:30)
    at f (/home/user/Documents/auth/node_modules/restify/node_modules/once/once.js:17:25)
    at Server.module.exports (/home/user/Documents/auth/routes/middleware/basic_auth.js:8:10)
    at next (/home/user/Documents/auth/node_modules/restify/lib/server.js:906:30)
    at f (/home/user/Documents/auth/node_modules/restify/node_modules/once/once.js:17:25)
    at parseJson (/home/user/Documents/auth/node_modules/restify/lib/plugins/json_body_parser.js:65:9)
    at Server.parseBody (/home/user/Documents/auth/node_modules/restify/lib/plugins/body_parser.js:90:13)
    at next (/home/user/Documents/auth/node_modules/restify/lib/server.js:906:30)
    at f (/home/user/Documents/auth/node_modules/restify/node_modules/once/once.js:17:25)
    at IncomingMessage.done (/home/user/Documents/auth/node_modules/restify/lib/plugins/body_reader.js:121:13)
    at IncomingMessage.g (events.js:260:16)
    at emitNone (events.js:67:13)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:905:12)
    at doNTCallback2 (node.js:441:9)
    at process._tickDomainCallback (node.js:396:17)
```

I might be doing something wrong, too, but given that the request will then timeout with no 500 returned, it's a bug. ;)

**Comment by moll** [comment]
This is at version f115867.

**Comment by Nicktho** [comment]
You have to include an entities `rev` property to be able to update it correctly.

Will need to handle this better.

**Comment by davidbanham** [comment]
Yeah, I noticed this in fixing up the test suite on driver. It's why #19 was written. I'll try and get this landed today.

**Comment by davidbanham** [comment]
Fixed with #19
